doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst

```diff
@@ -201,12 +201,26 @@ Automated metadata generation
 By enabling the ``SB_CONFIG_DFU_ZIP_BLUETOOTH_MESH_METADATA`` option in sysbuild, the metadata will be automatically parsed from the ``.elf`` and ``.config`` files.
 The parsed data is stored in the :file:`ble_mesh_metadata.json` file.
 The file is placed in the :file:`dfu_application.zip` archive in the build folder of the application.
-Additionally, the metadata string required by the ``mesh models dfu slot add`` command will be printed in the command line window when the application is built::
+Additionally, the metadata string and firmware ID required by the ``mesh models dfu slot add`` command is printed in the command-line window when the application is built::
 
   Bluetooth Mesh Composition metadata generated:
     Encoded metadata: 020000000000000094cf24017c26f3710100
+    Firmware ID: 59000200000000000000
     Full metadata written to: APPLICATION_FOLDER\build\zephyr\dfu_application.zip
 
+The Firmware ID can be generated from one of multiple sources:
+
+* To use a user-supplied hex-string as the Firmware ID, enable the ``SB_CONFIG_DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM`` option.
+  This is the default behavior.
+  * If this option is selected, the ``SB_CONFIG_DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM_HEX`` option must be set to a valid Firmware ID.
+    At minimum, it should include a Company ID in little-endian order.
+    The rest of the string is vendor-specific version information.
+  * When using the :ref:`zephyr:bluetooth_mesh_dfd_srv` for distribution, the number of bytes in the Firmware ID of images used during distribution must not exceed the value of the ``CONFIG_BT_MESH_DFU_FWID_MAXLEN`` option set on the distributor node.
+* To use a Firmware ID consisting of the value of the ``CONFIG_BT_COMPANY_ID`` option and the value of the ``CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION`` option, enable the ``SB_CONFIG_DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_MCUBOOT_VERSION`` option.
+
+.. note::
+   The Firmware ID inserted into the JSON file must match the ``fwid`` used when initializing the :ref:`zephyr:bluetooth_mesh_dfu_srv`.
+
```

Comment:
This note confuses me. When does the Firmware ID inserted into the JSON file must match?
1. At the time of first boot?
2. At the time of processing of Firmware Update Start? or in-between (like metadata checks, etc.)
3. At the time of attempting a confirm/verify stage or SWAP?

Can you please clarify this note?

Reply:
It's a bit difficult. The point is that, whenever you run "west build", any code that calls `BT_MESH_DFU_SRV_INIT(..., struct_with_pointer_to_fwid, ...)` and the value set in Kconfig for the `SB_CONFIG_DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_...` options should use the same FWID in those two locations. I don't know how to clarify this properly.

Reply:
don't know how this system works but why can't the sysbuild value be propagated to applications which then directly use it instead of having a potential mismatch?

Reply:
> don't know how this system works but why can't the sysbuild value be propagated to applications which then directly use it instead of having a potential mismatch?

Because we are making a minimal solution here.

Reply:
Can the value in the application be replaced with a Kconfig? If so I can knock up a PR to propagate the value from sysbuild to application in a couple of minutes

Reply:
Bit longer explanation: The API exposed to the applications currently is very open-ended and lets applications supply whatever byte string they want as FWID in the image list. To enforce a match between application behavior and JSON contents would mean several breaking API changes, which we need to think about before implementing.

Reply:
> Can the value in the application be replaced with a Kconfig? If so I can knock up a PR to propagate the value from sysbuild to application in a couple of minutes

Yes sure, but we can't force every existing or future application to use the Kconfig in their code. The API they have access to lets them do whatever they want.

Reply:
@omkar3141 I believe this is the only discussion left. Do you have more input after my explanation(s) above? I struggle to formulate this in an efficient and clear way.

Reply:
How about this? 
```suggestion
.. note::
   The Firmware ID (FWID) specified in the JSON file must match the ``fwid`` used when initializing the :ref:`zephyr:bluetooth_mesh_dfu_srv`. This ensures that after the firmware update is applied, the node will respond with the correct FWID in the Firmware Update Information Status message.

```

Reply:
And again, remember to have each sentence on a separate line.